GoogleForCreators/web-stories-wp
# Media: Clean up linked post meta on attachment deletion

## Context

If you delete a muted / optimized version of a video, also delete the post meta to link to the original video. 

## Summary



## Relevant Technical Choices

I created a util class so that the code can be shared. I thought about an abstract and traits to share code. A util class is what I needed up with. 

I would like to avoid doing a meta query, but I can't think of a way around it, other than a raw database query.   

## To-do



## User-facing changes



## Testing Instructions






This PR can be tested by following these steps:


1. Add video
2. Mute video
3. Permanently delete muted video from WP media library
4. Try to mute video again
5. See no error


## Reviews

### Does this PR have a security-related impact?



No

### Does this PR change what data or activity we track or use?



No

### Does this PR have a legal-related impact?



No

## Checklist




---



Fixes #12240
